runtime/near-vm-logic/src/logic.rs

```diff
@@ -24,6 +25,51 @@ pub type Result<T> = ::std::result::Result<T, VMLogicError>;
 const LEGACY_DEFAULT_PROTOCOL_VERSION: ProtocolVersion = 34;
 const IMPLICIT_ACCOUNT_CREATION_PROTOCOL_VERSION: ProtocolVersion = 35;
 
+pub fn transfer_exec_fee(
+    cfg: &ActionCreationConfig,
+    receiver_id: &AccountId,
+    current_protocol_version: ProtocolVersion,
+) -> Result<Gas> {
+    Ok(
+        if current_protocol_version >= IMPLICIT_ACCOUNT_CREATION_PROTOCOL_VERSION
+            && is_account_id_64_len_hex(&receiver_id)
+        {
+            cfg.create_account_cost
+                .exec_fee()
+                .checked_add(cfg.add_key_cost.full_access_cost.exec_fee())
+                .ok_or(HostError::IntegerOverflow)?
+                .checked_add(cfg.transfer_cost.exec_fee())
+                .ok_or(HostError::IntegerOverflow)?
+        } else {
+            cfg.transfer_cost.exec_fee()
+        },
+    )
```

Comment:
nit and mostly personal preference, but I usually write these as

```
    let res = if current_protocol_version >= IMPLICIT_ACCOUNT_CREATION_PROTOCOL_VERSION
        && is_account_id_64_len_hex(&receiver_id)
    {
        cfg.create_account_cost
            .exec_fee()
            .checked_add(cfg.add_key_cost.full_access_cost.exec_fee())
            .ok_or(HostError::IntegerOverflow)?
            .checked_add(cfg.transfer_cost.exec_fee())
            .ok_or(HostError::IntegerOverflow)?
    } else {
        cfg.transfer_cost.exec_fee()
    };
    Ok(res)
```

If you put a complex expression inside `Ok`  / `Some`, there's a chance that it'll grow *more* complex in the future, and you would want to pull parts of it out of `Ok`, which would shift the indentation. So, a more future-proof version is to extract the inner value into a variable from the start.

But, again, this is more "consider if you might like this alternative way", it not necessary better.

Reply:
Done